spinalcordtoolbox/scripts/sct_download_data.py

```diff
@@ -76,7 +76,9 @@
     },
     "binaries_linux": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/releases/download/r20220516-2/spinalcordtoolbox-binaries_linux.tar.gz",
+            #  "https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/releases/download/r20220516-2/spinalcordtoolbox-binaries_linux.tar.gz",
+            #  only for testing purposes
+            "https://syncandshare.lrz.de/dl/fi9eCRkTSJBH55a6qmpXc2k7/spinalcordtoolbox-binaries_linux.tar.gz",
         ],
         "default_location": __bin_dir__,
```

Comment:
This change caused an error:

```python
Trying URL: https://syncandshare.lrz.de/dl/fi9eCRkTSJBH55a6qmpXc2k7/spinalcordtoolbox-binaries_linux.tar.gz
Link download error, trying next mirror (error was: 'filename')
Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_download_data.py", line 241, in <module>
    main(sys.argv[1:])
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_download_data.py", line 234, in main
    install_data(url, dest_folder, keep=arguments.k)
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/download.py", line 138, in install_data
    tmp_file = download_data(url)
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/download.py", line 74, in download_data
    raise Exception('Download error', exceptions)
Exception: ('Download error', [KeyError('filename')])
```

Opening this comment just to mention that I'm looking into this now. :slightly_smiling_face:

Reply:
Hey @joshuacwnewton,

I noticed this as well, apparently I must have packaged the data a bit differently than the original sct folder -  however I did not have enough time to look into it again. I hope I can return to this later in the day. 

Thx!

Reply:
It looks like the issue stems from SCT's HTTP response parsing code:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2769d530209dc6e62afd32266417bfe4dc1a5897/spinalcordtoolbox/download.py#L41-L44

With the previous `github.com` URL, the value of `response.headers['Content-Disposition']` was:

```python
'attachment; filename=spinalcordtoolbox-binaries_linux.tar.gz'
```

Which gets parsed as this dict:

```python
{'filename': 'spinalcordtoolbox-binaries_linux.tar.gz'}
```

However, for the temporary `syncandshare.lrz.de` URL, the value of `response.headers['Content-Disposition']` is:

```python
"attachment;filename*=UTF-8''spinalcordtoolbox-binaries_linux.tar.gz"
```

Which gets parsed as this dict:

```python
{'filename*': "UTF-8''spinalcordtoolbox-binaries_linux.tar.gz"}
```

Hence the `KeyError` for `content['filename']` in the traceback.

Reply:
I'm not sure if this is an issue with how the website is formatting their HTTP response headers, or if this is a parsing issue on our end...

But, either way, I imagine the easiest way to resolve this would be to pick a different file hosting service?

Reply:
Another simple fix could be to do this:

```diff
-      filename = content["filename"] 
+     filename = content.get('filename', filename)
```

That way, if the `'filename'` key doesn't exist in the response header (due to malformed input or something), we can fall back on the original `filename` value.